homeassistant/components/whirlpool/climate.py

```diff
@@ -129,7 +129,7 @@ def __init__(
 
     async def async_added_to_hass(self) -> None:
         """Connect aircon to the cloud."""
-        self._aircon.register_attr_callback(self.async_write_ha_state)
+        await self._aircon.register_attr_callback(self.async_write_ha_state)
```

Comment:
According to the library, this method on the appliance is not a coroutine function.

https://github.com/abmantis/whirlpool-sixth-sense/blob/ae12505942b0b43cc3c9d4bb97e44ff0356e8bf1/whirlpool/appliance.py#L37

It looks like it's the test that mocks the library incorrectly.

https://github.com/home-assistant/core/blob/255611238bddd83b0f6949f31cdfd8ee359b4b6c/tests/components/whirlpool/conftest.py#L66

That should just be a `MagicMock` instead of an `AsyncMock`.

Reply:
Actually, the problem is in the test fixture - not sure why this is passing locally

Reply:
@MartinHjelmare @mkmer Made the change suggested above!

Reply:
Need to remove the added await calls

Reply:
There was another AsyncMock I had to change for those, but I have done it now.